### DISCUSSION TITLE
使用PAGImageView播放动画，动画会抖一下。

### DISCUSSION THREAD

**Discussion by 1079107009** [Tencent/libpag#2053] [Q&A]
源文件、缓存文件、视频：[data.zip](https://github.com/Tencent/libpag/files/13824710/data.zip)

**Comment by 1079107009** [comment]
[data.txt](https://github.com/Tencent/libpag/files/13825111/data.txt)
在第13帧的时候会出问题，是不是写入缓存的时候有点问题？

**Comment by 1079107009** [comment]
在写入缓存的时候会有几帧frameSize不一样导致的抖动，能不能帮忙看一下为什么会这样？
正常的情况：
SequenceFile::writeFrame(1) frameIndex = 4, frameSize = 412492, frameOffset = 14940717
SequenceFile::writeFrame(1) frameIndex = 5, frameSize = 412630, frameOffset = 15353221
SequenceFile::writeFrame(1) frameIndex = 9, frameSize = 414093, frameOffset = 17006930

抖动的情况：
SequenceFile::writeFrame(1) frameIndex = 4, frameSize = 411103, frameOffset = 410276
SequenceFile::writeFrame(1) frameIndex = 5, frameSize = 412492, frameOffset = 18252493
SequenceFile::writeFrame(1) frameIndex = 9, frameSize = 414410, frameOffset = 1648036

**Reply by kevingpqi123** [reply]
该问题应该出现在第一次渲染的过程中，且为部分中低端机型，首次渲染的过程中，如果渲染中会出现卡顿，是会出现跳帧渲染的问题，同时也会跳帧写入数据，缓存完成后续渲染的过程中就会正常

PAGImageView 就是为了解决这类场景，采用 PAGView 播放可能每次播放都会有该问题，而对于 PAGImageView 而言，如果磁盘缓存已经完成，后续渲染都是读取磁盘缓存数据，可以实现渲染性能与素材的无关性

如果想解决首次渲染的卡顿问题，有一种思路是提前缓存数据，根据实际渲染尺寸的大小，使用 PAGDecoder 的接口缓存每一帧数据， PAGDecoder 和 PAGImageView 使用的磁盘缓存是同一份
